.github/workflows/test.yaml

```diff
@@ -32,6 +32,40 @@ jobs:
           ALL_CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files }}
         run: bun ./ci/index.ts
 
+  shell-format-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Install shfmt
+        run: |
+          wget -O shfmt https://github.com/mvdan/sh/releases/download/v3.8.0/shfmt_v3.8.0_linux_amd64
+          chmod +x shfmt
+          sudo mv shfmt /usr/local/bin/
+      - name: Check shell script formatting
+        run: shfmt -d .
+
+  shell-script-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Install shellcheck
+        run: |
+          wget -O shellcheck.tar.xz https://github.com/koalaman/shellcheck/releases/download/v0.9.0/shellcheck-v0.9.0.linux.x86_64.tar.xz
+          tar -xf shellcheck.tar.xz
+          sudo mv shellcheck-v0.9.0/shellcheck /usr/local/bin/
+      - name: Check shell script quality
+        run: |
+          # -S error: Only fail on errors, not warnings
+          # -e SC1091: Ignore missing source warning
+          # {} +: Pass multiple files to shellcheck at once for efficiency
+          find . -name "*.sh" \
+            -not -path "./.git/*" \
+            -not -path "./runtimes/.test/*" \
+            -not -path "./node_modules/*" \
```

Comment:
```suggestion
            -not -path "./**/node_modules/*" \
```

Current command doesnt ignore it well, doing this should help, as seen here:

<img width="2388" height="1038" alt="Image" src="https://github.com/user-attachments/assets/5f01f484-10bb-4b22-a7c8-2845cac77df1" />

Reply:
Same in contributing guide